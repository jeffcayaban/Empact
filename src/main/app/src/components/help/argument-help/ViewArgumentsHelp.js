import React from "react";
import {Col, Grid, Row} from "react-bootstrap";
import ArgumentEntryDiagram from '../../../other/images/argument-entry-diagram.png';
import ArgumentPageDiagram from '../../../other/images/argument-page-diagram.png';
import ArgumentVisualisationBtn from '../../../other/images/argument-visualisation-btn.png';
import ArgumentVisualisation from '../../../other/images/argument-visualisation.png';
import BackButton from "../../common/BackButton";

/**
 * Displays help information for viewing arguments.
 */

class ViewArgumentsHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Viewing Arguments</h2>
                    <p>Understand how arguments are viewed.</p>
                </div>
                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <div>
                                    <h4 style={{ margin: 0 }}>Viewing arguments on a petition or argument</h4>
                                    <hr className={"helpHr"} />
                                    <p>
                                        When you are on a petition or argument, you may sometimes notice that there
                                        are a set of arguments that are located on the lower part of the page. The
                                        image below explains the different things you may see for a set of supporting
                                        arguments.
                                    </p>
                                    <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                        <img style={{ maxWidth: '70%', minWidth: '350px', align: 'center' }}
                                             src={ArgumentEntryDiagram}
                                             alt="Argument Entry Diagram" />
                                    </div>
                                </div>
                                <div style={{ paddingTop: '35px' }}>
                                    <h4 style={{ margin: 0 }}>Viewing an argument on its page</h4>
                                    <hr className={"helpHr"} />
                                    <p>
                                        The image below explains the different things you may see when you are on an
                                        argument page.
                                    </p>
                                    <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                        <img style={{ maxWidth: '70%', minWidth: '350px', align: 'center' }}
                                             src={ArgumentPageDiagram}
                                             alt="Argument Page Diagram" />
                                    </div>
                                </div>
                                <div style={{ paddingTop: '35px' }}>
                                    <h4 style={{ margin: 0 }}>Viewing the tree of arguments for an argument</h4>
                                    <hr className={"helpHr"} />
                                    <p>
                                        Click on the "Arguments Overview" button found within the page of an
                                        argument to view a data visualisation of the arguments that were made to
                                        support or oppose the argument.
                                    </p>
                                    <div className={"centerStyle"} style={{ paddingTop: '15px', paddingBottom: '20px' }}>
                                        <img style={{ width: '180px', align: 'center' }}
                                             src={ArgumentVisualisationBtn}
                                             alt="Argument Page Diagram" />
                                    </div>
                                    <p>
                                        A diagram showing the arguments supporting and opposing the argument is shown
                                        after clicking the button. A red circle shows that there exists an argument
                                        that is opposing argument above it. A green circle shows that there exists an
                                        argument that is supporting the argument above it. Clicking on any of the circles
                                        will send you to that argument's page.
                                    </p>
                                    <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                        <img style={{ maxWidth: '25%', minWidth: '350px', align: 'center' }}
                                             src={ArgumentVisualisation}
                                             alt="Argument Data Visualisation" />
                                    </div>
                                </div>
                                <BackButton link={"/help/argument"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default ViewArgumentsHelp;