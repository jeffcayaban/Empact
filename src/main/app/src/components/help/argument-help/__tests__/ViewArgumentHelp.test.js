import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import ViewArgumentsHelp from "../ViewArgumentsHelp";

describe('ViewArgumentsHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<ViewArgumentsHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
