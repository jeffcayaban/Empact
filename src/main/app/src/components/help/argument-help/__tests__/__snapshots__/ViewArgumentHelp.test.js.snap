// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewArgumentsHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Viewing Arguments
    </h2>
    <p>
      Understand how arguments are viewed.
    </p>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <div>
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              Viewing arguments on a petition or argument
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              When you are on a petition or argument, you may sometimes notice that there are a set of arguments that are located on the lower part of the page. The image below explains the different things you may see for a set of supporting arguments.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "15px",
                }
              }
            >
              <img
                alt="Argument Entry Diagram"
                src="argument-entry-diagram.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "70%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
          </div>
          <div
            style={
              Object {
                "paddingTop": "35px",
              }
            }
          >
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              Viewing an argument on its page
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              The image below explains the different things you may see when you are on an argument page.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "15px",
                }
              }
            >
              <img
                alt="Argument Page Diagram"
                src="argument-page-diagram.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "70%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
          </div>
          <div
            style={
              Object {
                "paddingTop": "35px",
              }
            }
          >
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              Viewing the tree of arguments for an argument
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              Click on the "Arguments Overview" button found within the page of an argument to view a data visualisation of the arguments that were made to support or oppose the argument.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "15px",
                }
              }
            >
              <img
                alt="Argument Page Diagram"
                src="argument-visualisation-btn.png"
                style={
                  Object {
                    "align": "center",
                    "width": "180px",
                  }
                }
              />
            </div>
            <p>
              A diagram showing the arguments supporting and opposing the argument is shown after clicking the button. A red circle shows that there exists an argument that is opposing argument above it. A green circle shows that there exists an argument that is supporting the argument above it. Clicking on any of the circles will send you to that argument's page.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "15px",
                }
              }
            >
              <img
                alt="Argument Data Visualisation"
                src="argument-visualisation.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "25%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
          </div>
          <BackButton
            link="/help/argument"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
